### PR TITLE
tests: benchmarks: move pthread_pressure to benchmarks/posix

### PR DESCRIPTION
**Pull request by cfriedt**
The pthread_pressure test was not a typical test per se. It was a benchmark in search of the proper home.

Let's move it to the correct place in the Zephyr tree, add a doc, and provide some reporting.

Currently, k_threads out-perform pthreads by almost a factor of 2. The theoretical maximum performance of pthreads would be at parity of k_threads, since pthreads are a wrapper around kernel threads. It would be great to reduce the gap.

### CODE REVIEW

File: tests/benchmarks/posix/threads/README.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- CONFIG_TEST_PTHREADS - Exercise pthreads in the test app.
+- CONFIG_TEST_STACK_SIZE - Size of each thread stack in this test.
+
+The following table summarizes the purposes of the different extra
+configuration files that are available to be used with this benchmark.
+A tester may mix and match them allowing them different scenarios to
+be easily compared the default.
+
++-----------------------------+----------------------------------------+
+| prj-assert.conf             | Enable assertions for API verification |                 |
```
Comment (ycsin) [inline tests/benchmarks/posix/threads/README.rst:44]:
```suggestion
| prj-assert.conf             | Enable assertions for API verification |
```

File: tests/benchmarks/posix/threads/README.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    *** Booting Zephyr OS build v4.0.0-1410-gfca33facee37 ***
+    ASSERT: y
+    BOARD: qemu_riscv64
+    NUM_CPUS: 1
+    TEST_DELAY_US: 0
+    TEST_DURATION_S: 5
+    SMP: n
+    API, Thread ID, time(s), threads, cores, rate (threads/s/core)
+    k_thread, ALL, 5, 47663, 1, 9532
+    pthread, ALL, 5, 28180, 1, 5636
```
Comment (ycsin) [inline tests/benchmarks/posix/threads/README.rst:23]: I think an efficiency metric (# of `pthread`s ÷ # of `k_thread`s x 100%) here would be nice

**Review by ycsin — DISMISSED** [review]
LGTM

File: tests/benchmarks/posix/threads/README.rst (hunk 2 above)
Comment (cfriedt) [reply tests/benchmarks/posix/threads/README.rst:23]: Probably. The record regex is very close to the 100-character limit in yaml though, so I will probably leave that for a later improvement.

**Comment by cfriedt** [comment]
I guess technically, with assertions enabled, this would become a stress test rather than a benchmark.

Any thoughts about how that should be organized @peter-mitsis ?

**Review by ycsin — APPROVED** [review]

**Review by peter-mitsis — APPROVED** [review]

**Comment by peter-mitsis** [comment]
> Any thoughts about how that should be organized?

I don't have anything new to offer/suggest; I'm good with how this has been laid out.

**Pull request merged**